engine/beef_tx_bytes.go

```diff
@@ -55,26 +56,26 @@ func (beefTx *beefTx) toBeefBytes() ([]byte, error) {
 	return buffer, nil
 }
 
-func toBeefBytes(tx *bt.Tx, bumps BUMPs) []byte {
+func toBeefBytes(tx *trx.Transaction, bumps BUMPs) []byte {
 	txBeefBytes := tx.Bytes()
 
 	bumpIdx := getBumpPathIndex(tx, bumps)
 	if bumpIdx > -1 {
 		txBeefBytes = append(txBeefBytes, hasBUMP)
-		txBeefBytes = append(txBeefBytes, bt.VarInt(bumpIdx).Bytes()...)
+		txBeefBytes = append(txBeefBytes, trx.VarInt(bumpIdx).Bytes()...)
 	} else {
 		txBeefBytes = append(txBeefBytes, hasNoBUMP)
 	}
 
 	return txBeefBytes
 }
 
-func getBumpPathIndex(tx *bt.Tx, bumps BUMPs) int {
+func getBumpPathIndex(tx *trx.Transaction, bumps BUMPs) int {
 	bumpIndex := -1
 
 	for i, bump := range bumps {
 		for _, path := range bump.Path[0] {
-			if path.Hash == tx.TxID() {
+			if path.Hash == tx.TxID().String() {
```

Comment:
Generally the `TxID` and its `String` methods are tricky. Because that are not computentially `light`. They actually recalculate every time they are called. 

That's why I'm suggesting to store the `txID` in temporary variables and reuse it in the loop. 
```golang
txID := tx.TxID().String()
	
	for i, bump := range bumps {
		for _, path := range bump.Path[0] {
			if path.Hash == txID {
				bumpIndex = i
			}
		}
	}
```

For the record, this is implementation of `TxID`:
```golang
func (tx *Transaction) TxID() *chainhash.Hash {
	txid, _ := chainhash.NewHash(crypto.Sha256d(tx.Bytes()))
	return txid
}
```

And the `String`:
```golang
func (hash Hash) String() string {
	for i := 0; i < HashSize/2; i++ {
		hash[i], hash[HashSize-1-i] = hash[HashSize-1-i], hash[i]
	}
	return hex.EncodeToString(hash[:])
}
```